### PR TITLE
spring_boot_security_authentication_password_encrypt_and_logout

### PR DESCRIPTION
**Pull request by G-jonathan**
**- Set up user authentication with spring security.**
_(SpringSecurityConfig.configure)_
**- Creation of a table authorities to manage roles, this is needed by spring security operation.**
`CREATE TABLE public.authorities (`
    `email     VARCHAR(254) NOT NULL,`
    `authority VARCHAR(50)  NOT NULL,`
    `CONSTRAINT authorities_pk PRIMARY KEY (email)`
`);`
`ALTER TABLE public.authorities ADD CONSTRAINT user_account_authorities_fk`
    `FOREIGN KEY (email)`
        `REFERENCES public.user_Account (email)`
        `ON DELETE NO ACTION`
        `ON UPDATE NO ACTION`
        `NOT DEFERRABLE;`
**- Modification of user passwords in the database :  encryption with BCrypt (strong, 10 round)**
- screen_password_encrypt
**- Implementation of the logout logic** 
_(SecurityService.logout)_

Card Link -->> https://github.com/G-jonathan/PayMyBuddy/projects/6#card-73972856

### CODE REVIEW

File: src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.server.VaadinServletRequest;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.logout.SecurityContextLogoutHandler;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SecurityService {
```
Comment (Asavarkhul) [inline src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java]: I don't like the Service name, I would prefer to put sth like `SecurityProvider` but it's more a pov than an advice :)

File: src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+    private static final String LOGOUT_SUCCESS_URL = "/";
+
+    public UserDetails getAuthenticatedUser() {
+        SecurityContext context = SecurityContextHolder.getContext();
```
Comment (Asavarkhul) [inline src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java]: This object `SecurityContext` should have been injected directly on `SecurityService` initialisation

File: src/main/resources/SQL_Scripts/database_creation_script.sql [hunk 3]
```diff
@@ -1,120 +1,134 @@
...
 CREATE TABLE public.bank_account (
-                id INTEGER NOT NULL DEFAULT nextval('public.bank_account_id_seq'),
-                holder_first_name VARCHAR(100) NOT NULL,
-                holder_last_name VARCHAR(100) NOT NULL,
-                iban VARCHAR(31) NOT NULL,
-                bic VARCHAR(11) NOT NULL,
-                user_account_email VARCHAR(254) NOT NULL,
-                CONSTRAINT bank_account_pk PRIMARY KEY (id)
+    id                 INTEGER      NOT NULL DEFAULT nextval('public.bank_account_id_seq'),
+    holder_first_name  VARCHAR(100) NOT NULL,
```
Comment (Asavarkhul) [inline src/main/resources/SQL_Scripts/database_creation_script.sql:21]: Does 100 characters being enough? I mean, in this case yes surely, but aren't we used to put sth more like 255?

**Review by Asavarkhul — APPROVED** [review]

File: src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java (hunk 1 above)
Comment (G-jonathan) [reply src/main/java/com/openclassroomsproject/paymybuddy/configuration/security/SecurityService.java]: done, name changed.

File: src/main/resources/SQL_Scripts/database_creation_script.sql (hunk 3 above)
Comment (G-jonathan) [reply src/main/resources/SQL_Scripts/database_creation_script.sql:21]: we can discuss it tomorrow so :)

**Pull request merged**